### PR TITLE
Update api lint checks for migration-tools

### PR DESCRIPTION
**Pull request by ChumpChief**
Attempts to fix api linting for the migration-tools package.  It kind of works, but I'm still seeing some oddities in task order, want to see if this will work in CI.

[AB#13807](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/13807)

### CODE REVIEW

File: examples/utils/migration-tools/package.json [hunk 1]
```diff
@@ -36,13 +46,19 @@
...
 	},
 	"scripts": {
+		"api-extractor:commonjs": "flub generate entrypoints --outDir ./dist",
 		"api-extractor:esnext": "flub generate entrypoints --outDir ./lib",
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
 		"build:esnext": "tsc --project ./tsconfig.json",
 		"check:are-the-types-wrong": "echo skip attw --pack .",
 		"check:biome": "biome check .",
-		"check:exports": "api-extractor run --config ./api-extractor-lint.json",
```
Comment (jason-ha) [inline examples/utils/migration-tools/package.json:45]: Should be able to delete `api-extractor-lint.json`.
And also `api-extractor.json` that was already unused.

**Review by jason-ha — APPROVED** [review]
Does this have public types?
(Feels better to me not to have those exports if there aren't any, but no biggie.)

File: examples/utils/migration-tools/package.json (hunk 1 above)
Comment (ChumpChief) [reply examples/utils/migration-tools/package.json:45]: Removed in latest

**Comment by ChumpChief** [comment]
> Does this have public types?
> (Feels better to me not to have those exports if there aren't any, but no biggie.)

Longer-term it will, so figured it would make sense to go ahead and prep for it.

**Comment by jason-ha** [comment]
> > Does this have public types?
> > (Feels better to me not to have those exports if there aren't any, but no biggie.)
> 
> Longer-term it will, so figured it would make sense to go ahead and prep for it.

Won't types go to beta after alpha, before public? We have all the things that should make the transitions to more supported easy. An advantage of not public now is that you can't accidentally use it. For example, this package doesn't have API reports turned on, which would be the way to see that public exposure is happening.

**Comment by ChumpChief** [comment]
> > > Does this have public types?
> > > (Feels better to me not to have those exports if there aren't any, but no biggie.)
> > 
> > 
> > Longer-term it will, so figured it would make sense to go ahead and prep for it.
> 
> Won't types go to beta after alpha, before public? We have all the things that should make the transitions to more supported easy. An advantage of not public now is that you can't accidentally use it. For example, this package doesn't have API reports turned on, which would be the way to see that public exposure is happening.

OK! Updated in latest

**Pull request merged**